### PR TITLE
[bitnami/mongodb] Fixed NodePort External Access

### LINKED ISSUE #5508: [bitnami/mongodb] ReplicaSet not initialized when using NodePort and externalAccess is enabled
**Which chart**:
bitnami/mongodb 10.7.0

**Describe the bug**
When MongoDB tries to initialize a replicaset it uses the value set in $MONGODB_ADVERTISED_HOSTNAME. Using NodePort  for external access results in
```

WARN  ==> Problem initiating replica set
            request: rs.initiate({"_id":"rs0", "members":[{"_id":0,"host":"mongodb.k8s-dev.example.com:27017","priority":5}]})
            response: MongoDB shell version v4.4.4
connecting to: mongodb://127.0.0.1:27017/admin?compressors=disabled&gssapiServiceName=mongodb
Implicit session: session { "id" : UUID("67002c19-b3c3-4beb-ae22-25dcf9fd89d0") }
MongoDB server version: 4.4.4
{
	"ok" : 0,
	"errmsg" : "No host described in new configuration with {version: 1, term: 0} for replica set rs0 maps to this node",
	"code" : 93,
	"codeName" : "InvalidReplicaSetConfig"
}
```

Regardless of using autodiscovery or not, setting externalAccess.enabled=true and using NodePort won't create a working mongo replicaset. externalAccess.enabled=false will create a working replicaset but there is no way to access it from outside our Kubernetes.

**To Reproduce**

1. install the mongodb Helm Chart using this values.yaml:
```yaml
fullnameOverride: testdb

replicaCount: 3

architecture: replicaset

auth:
  enabled: true
  rootPassword: "root"
  database: "demodb"
  username: "demouser"
  password: "password123"
  replicaSetKey: "xndbbu2x0f"

arbiter:
  enabled: true
  limits:
    cpu: 100m
    memory: 128Mi
  requests:
    cpu: 100m
    memory: 128Mi

persistence:
  enabled: true
  storageClass: "rbd"

externalAccess:
  enabled: true
  autoDiscovery:
    enabled: true
  service:
    type: "NodePort"
    domain: "mongodb.k8s-dev.example.com"
    nodePorts:
      - "30210"
      - "30211"
      - "30212"

serviceAccount:
  create: false

rbac:
  create: true

metrics:
  enabled: false

pdb:
  create: false

resources:
  requests:
    memory: 512Mi
  limits:
    memory: 512Mi

```
helm upgrade --install demo-mongo bitnami/mongodb -f test.yaml

2. kubectl logs -f testdb-0 will show the rs initializing error above after the mongo has configured itself

**Expected behavior**
I would expect the mongoDB initializes correctly and will be accessible from within the cluster and from outside.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.1", GitCommit:"fe51cd1e31e6a202cba7dead9552a6d418ded79a", GitTreeState:"clean", GoVersion:"go1.13.10"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:23:21Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
Our Kubernetes runs on premise and the mongodb will only be accessed from within our private network. Not setting a value in externalAccess.service.domain will result in an ip lookup with our external ip which won't work. I've also tried using the domain name and ip of one of the nodes which also won't work. The domain specified in the values.yaml will be forwarded from our Netscaler towards the Kubernetes cluster.

### PR DESCRIPTION
**Pull request by deveshk0**
**Description of the change**

Fixed Node Port External Access

**Benefits**

- Ability to run mongo cluster on nodePort

**Possible drawbacks**

NA

**Applicable issues**

  - fixes #5508

**Additional information**

⚠️ This PR depends on this PR - https://github.com/bitnami/bitnami-docker-mongodb/pull/311

**Checklist** 

- [ X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ X] Variables are documented in the README.md
- [ X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Comment by deveshk0** [comment]
@carrodher @juan131 Can you Please have a look into this

File: bitnami/mongodb/templates/replicaset/scripts-configmap.yaml [hunk 1]
```diff
@@ -87,6 +89,7 @@ data:
     {{- end }}
 
     echo "Advertised Hostname: $MONGODB_ADVERTISED_HOSTNAME"
+    echo "Advertised Port: $MONGODB_ADVERTISED_PORT_NUMBER"
```
Comment (migruiz4) [inline bitnami/mongodb/templates/replicaset/scripts-configmap.yaml:92]: This comment should also be added in line 145.

File: bitnami/mongodb/templates/replicaset/scripts-configmap.yaml [hunk 2]
```diff
@@ -74,7 +74,9 @@ data:
     {{- if .Values.externalAccess.service.domain }}
     export MONGODB_ADVERTISED_HOSTNAME={{ .Values.externalAccess.service.domain }}
     {{- else }}
-    export MONGODB_ADVERTISED_HOSTNAME=$(curl -s https://ipinfo.io/ip)
+    ID="${MY_POD_NAME#"{{ $fullname }}-"}"
+    export MONGODB_ADVERTISED_HOSTNAME=$MY_POD_HOST_IP
+    export MONGODB_ADVERTISED_PORT_NUMBER=$(echo '{{ .Values.externalAccess.service.nodePorts }}' | tr -d '[]' | cut -d ' ' -f "$(($ID + 1))")
```
Comment (migruiz4) [inline bitnami/mongodb/templates/replicaset/scripts-configmap.yaml:79]: Why would `MONGODB_ADVERTISED_PORT_NUMBER` be conditioned to `externalAccess.service.domain` not being set?

```suggestion
    ID="${MY_POD_NAME#"{{ $fullname }}-"}"
    export MONGODB_ADVERTISED_PORT_NUMBER=$(echo '{{ .Values.externalAccess.service.nodePorts }}' | tr -d '[]' | cut -d ' ' -f "$(($ID + 1))")
    {{- if .Values.externalAccess.service.domain }}
    export MONGODB_ADVERTISED_HOSTNAME={{ .Values.externalAccess.service.domain }}
    {{- else }}
    export MONGODB_ADVERTISED_HOSTNAME=$MY_POD_HOST_IP
```

File: bitnami/mongodb/templates/replicaset/scripts-configmap.yaml [hunk 3]
```diff
@@ -125,7 +128,9 @@ data:
     {{- if .Values.externalAccess.hidden.service.domain }}
     export MONGODB_ADVERTISED_HOSTNAME={{ .Values.externalAccess.hidden.service.domain }}
     {{- else }}
-    export MONGODB_ADVERTISED_HOSTNAME=$(curl -s https://ipinfo.io/ip)
+    ID="${MY_POD_NAME#"{{ $fullname }}-hidden-"}"
+    export MONGODB_ADVERTISED_HOSTNAME=$MY_POD_HOST_IP
+    export MONGODB_ADVERTISED_PORT_NUMBER=$(echo '{{ .Values.externalAccess.hidden.service.nodePorts }}' | tr -d '[]' | cut -d ' ' -f "$(($ID + 1))")
```
Comment (migruiz4) [inline bitnami/mongodb/templates/replicaset/scripts-configmap.yaml:133]: Same as mentioned above.

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Hi @deveshk0,

Thank you very much for your contribution! Could you please have a look at my comments below?

File: bitnami/mongodb/templates/replicaset/scripts-configmap.yaml (hunk 1 above)
Comment (deveshk0) [reply bitnami/mongodb/templates/replicaset/scripts-configmap.yaml:92]: Thanks,
Added in line 145

File: bitnami/mongodb/templates/replicaset/scripts-configmap.yaml (hunk 2 above)
Comment (deveshk0) [reply bitnami/mongodb/templates/replicaset/scripts-configmap.yaml]: yes, you are correct, my bad. Updated

**Comment by migruiz4** [comment]
Thank you @deveshk0 for applying the suggested changes! Your PR looks good to me, but we will need to wait for https://github.com/bitnami/bitnami-docker-mongodb/pull/311 to be approved and merged before we can merge this PR.

**Comment by deveshk0** [comment]
@migruiz4 Any blocker here?

**Review by migruiz4 — APPROVED** [review]
LGTM! The PR may take a bit longer to merge since it will be merged by the bitnami_bot to ensure the container image is updated.

**Comment by sunxunkang** [comment]
> LGTM！PR 可能需要更长的时间来合并，因为它将由 bitnami_bot 合并以确保更新容器映像。
Has the whole problem been fixed? I have the same problem on bitnami mongodb chart version 10.31.5

**Comment by marcosbc** [comment]
@sunxunkang If you continue finding issues, I encourage you to open a new issue so we can understand further what might be happening in your case. Please also make sure to link to this PR, just in case.

**Pull request merged**